syntax = "proto3";

package onyx_otc.v1;
option csharp_namespace = "Onyx.Otc";


enum OtcErrorCode {
  OTC_ERROR_CODE_UNSPECIFIED = 0;
  OTC_ERROR_CODE_INVALID_REQUEST = 1;
  OTC_ERROR_CODE_NOT_IMPLEMENTED = 2;
  OTC_ERROR_CODE_UNAUTHENTICATED = 3;
  OTC_ERROR_CODE_TOO_MANY_REQUESTS = 4;
}

enum Side {
  SIDE_UNSPECIFIED = 0;
  SIDE_BUY = 1;
  SIDE_SELL = 2;
}

enum SubscriptionStatus {
  SUBSCRIPTION_STATUS_UNSPECIFIED = 0;
  SUBSCRIPTION_STATUS_SUBSCRIBED = 1;
  SUBSCRIPTION_STATUS_UNSUBSCRIBED = 2;
}

enum Method {
  METHOD_UNSPECIFIED = 0;
  METHOD_AUTH = 1;
  METHOD_ORDER = 2;
  METHOD_SUBSCRIBE = 3;
  METHOD_UNSUBSCRIBE = 4;
}


enum Exchange {
  EXCHANGE_UNSPECIFIED = 0;
  EXCHANGE_ICE = 1;
  EXCHANGE_CME = 2;
}


enum OrderType {
  ORDER_TYPE_UNSPECIFIED = 0;
  ORDER_TYPE_FILL_OR_KILL = 1;
}


enum OrderState {
  ORDER_STATE_UNSPECIFIED = 0;
  ORDER_STATE_PENDING_CREATED = 1;
  ORDER_STATE_CREATED = 2;
  ORDER_STATE_NEW = 3;
  ORDER_STATE_FILLED = 6;
  ORDER_STATE_ERRORED = 10;
}

enum TradingError {
  TRADING_ERROR_UNSPECIFIED = 0;
  TRADING_ERROR_FORBIDDEN = 403;
  TRADING_ERROR_NOT_FOUND = 404;
  TRADING_ERROR_TRADING_DISABLED = 405;
  TRADING_ERROR_TRADING_SUSPENDED = 406;
  TRADING_ERROR_NOT_SUBSCRIBED = 407;
  TRADING_ERROR_INVALID_SYMBOL = 413;
  TRADING_ERROR_INVALID_VALUE = 422;
  TRADING_ERROR_INVALID_STATE = 424;
  TRADING_ERROR_INVALID_PRICE = 425;
  TRADING_ERROR_CONTRACT_EXPIRING = 426;
  TRADING_ERROR_INTERNAL_ERROR = 500;
  TRADING_ERROR_NOT_IMPLEMENTED = 501;
}

enum Channel {
  CHANNEL_UNSPECIFIED = 0;
  CHANNEL_ORDERS = 1;
}
